Fixes so it works correctly on Spigot/Paper 1.20.1

tadhunt opened the pull request:
* Fixed an issue with exceptions generated when signs are not inside GriefPrevention claims
* Fixed an issue where the text on [sell] signs wasn't showing up
  * Updated to Spigot API 1.20.1 (to fix sell signs)
  * Put sign text on front and back of signs
* Fixed null warnings
* Fixed build (which required updating the maven shade plugin)

EtienneDx commented:
I see this is based on 1.5.0 which I never officially released. I won't have time to properly test this, could you provide me with the built jars so that I can add a new release?

I'll also leave you a couple comments on the code for things that'd be nice to change (eg changing to 1.5.0-pre2)

Thanks a lot for the contribution, glad to see it's still running

EtienneDx commented on pom.xml:
```diff
@@ -4,7 +4,7 @@
 	<groupId>Me.EtienneDx</groupId>
 	<artifactId>real-estate</artifactId>
-	<version>1.5.0-pre1</version>
+	<version>1.5.0-pre1-MCPD1</version>
```
```suggestion
	<version>1.5.0-pre2</version>

```

EtienneDx commented on src/me/EtienneDx/RealEstate/Transactions/ClaimAuction.java:
```diff
@@ -110,10 +112,16 @@ else if(!Utils.makePayment(owner, null, price, false, false))
-                        sign.setLine(1, Messages.getMessage(RealEstate.instance.messages.msgSignAuctionWon, false));
-                        sign.setLine(2, buyerPlayer.getName());
-                        sign.setLine(3, "");
+                        SignSide front = sign.getSide(Side.FRONT);
+                        SignSide back = sign.getSide(Side.BACK);
+                        front.setLine(0, "");
+                        front.setLine(1, Messages.getMessage(RealEstate.instance.messages.msgSignAuctionWon, false));
+                        front.setLine(2, buyerPlayer.getName());
+                        front.setLine(3, "");
+                        back.setLine(0, "");
+                        back.setLine(1, Messages.getMessage(RealEstate.instance.messages.msgSignAuctionWon, false));
+                        back.setLine(2, buyerPlayer.getName());
+                        back.setLine(3, "");
```
if you have some time, would be cool to have something like setLine(front, back, "stuff"). Otherwise that works (there's already a bazillion things to refactor haha ^^')

tadhunt replied:
Done.  I created a RealEstateSign class that encapsulates this and "isRealEstateSign" functionality.

EtienneDx reviewed (commented):
Yep all looks good, if you have some time to refactor writing to the sign that'd be great but otherwise I'm ok with it.
Would also be great to keep the versionning, either by going to 1.5.0-pre2, or full release if it's ready (not sure of the state of the project, haven't had time in years).

tadhunt commented:
You can find the jar here: https://github.com/tadhunt/RealEstate/releases/tag/1.5.0-pre2

EtienneDx reviewed (commented):
Love the refactor, I just saw one s.update that's been removed, is that voluntary? If it is all good to merge :)

tadhunt commented on src/me/EtienneDx/RealEstate/Transactions/ClaimAuction.java:
```diff
@@ -109,24 +110,23 @@ else if(!Utils.makePayment(owner, null, price, false, false))
-                sign.setLine(1, Messages.getMessage(RealEstate.instance.messages.msgSignAuctionEnded, false));
-                sign.setLine(2, "");
-                sign.setLine(3, "");
-                sign.update();
```
whoops, good catch.  That was a real mistake, I didn't mean to remove it.  It's added back now.

tadhunt commented:
I fixed the missing s.update(), and added a new 1.5.0-pre2 release built against this branch: https://github.com/tadhunt/RealEstate/releases/tag/1.5.0-pre2

I'll also submit a PR to add a github actions build workflow so that you don't need to trust my binaries :)

EtienneDx commented:
That's perfect, thanks a lot for all that!
